File: BubbleSort/CPP/BubbleSortTest.cpp
#include "include/BubbleSort.hpp"
#include "include/BubbleSortTest.hpp"
#include<iostream>



bool BubbleSortTest::test(std::vector<int>& input, std::vector<int>& output){
    BubbleSort bubbleSort;
    std::vector<int> res = bubbleSort.bubbleSort(input);
    if(res.size()!=output.size()){
      return false;
    }
    for(int i=0;i<res.size();i++){
      if(res[i]!=output[i]){
        return false;
      }
    }
    return true;
}

File: BubbleSort/JAVA/BubbleSortTest.java


public class BubbleSortTest {
  
  boolean test(int[] input,int[] expected_output){
    BubbleSort bs = new BubbleSort();
    int[] output = bs.bubbleSort(input);
    if(output.length != input.length) return false;
    for(int i=0;i<output.length;i++){
      if(output[i]!=expected_output[i])
        return false;
    }
    System.out.println();
    return true;
  }
}
